src/nvme/ioctl.h

```diff
@@ -2157,12 +2159,67 @@ static inline int nvme_get_log_lockdown(nvme_link_t l,
 /**
  * nvme_set_features() - Set a feature attribute
  * @l:		Link handle
- * @args:	&struct nvme_set_features_args argument structure
+ * @result:	The command completion result from CQE dword0
+ * @data:	User address of feature data, if applicable
+ * @timeout:	Timeout in ms
+ * @nsid:	Namespace ID, if applicable
+ * @cdw11:	Value to set the feature to
+ * @cdw12:	Feature specific command dword12 field
+ * @cdw13:	Feature specific command dword13 field
+ * @cdw15:	Feature specific command dword15 field
+ * @data_len:	Length of feature data, if applicable, in bytes
+ * @save:	Save value across power states
+ * @uuidx:	UUID Index for differentiating vendor specific encoding
+ * @fid:	Feature identifier
  *
  * Return: 0 on success, the nvme command status if a response was
  * received (see &enum nvme_status_field) or a negative error otherwise.
  */
-int nvme_set_features(nvme_link_t l, struct nvme_set_features_args *args);
+static inline int nvme_set_features(nvme_link_t l, __u32 *result, void *data,
+				     __u32 timeout, __u32 nsid, __u32 cdw11,
+				     __u32 cdw12, __u32 cdw13, __u32 cdw15,
+				     __u32 data_len, bool save, __u8 uuidx,
+				     __u8 fid)
```

Comment:
Just a quick review I'll look into to it more tomorrow.

Drop the timeout argumen, in the most cases it is the default anyway. if a user want's to use a different timeout, it should be done via the link object.

Also move the result argument as last, it's the return value.

Reply:
okay, so you are saying we should do something like

```c
.timeout_ms	= nvme_link_get_timeout(l),
```

?

I was thinking if the submitting command has a timeout value of 0 the passthru function just uses the value from the link. This would avoid a lot of duplicated code on this level.

Reply:
After some iterations I ended up doing what you were thinking of :)